mqtt_client.c

```diff
@@ -990,6 +992,30 @@ static esp_err_t deliver_publish(esp_mqtt_client_handle_t client)
     return ESP_OK;
 }
 
+static esp_err_t deliver_suback(esp_mqtt_client_handle_t client)
+{
+    uint8_t *msg_buf = client->mqtt_state.in_buffer;
+    size_t msg_total_len = client->mqtt_state.message_length;
+    // SUBACK payload length = total length - (fixed header (2 bytes) + variable header (2 bytes))
+    int total_data_len = msg_total_len - 4;
+    char *msg_data = (char *)&msg_buf[4];
```

Comment:
Same as above, suggest creating an inline method to calculate the payload pointer, such as
```cpp
static inline char *mqtt_get_suback_data(uint8_t *buffer, size_t *length)
{
    // SUBACK payload length = total length - (fixed header (2 bytes) + variable header (2 bytes))
   ...
}
```
and moving it to `mqtt_msg.h`